### PR TITLE
Add a Metadata-Flavor HTTP header to the initial metadata request

### PR DESCRIPTION
**Pull request by mareksuscak**
An attempt to fix https://github.com/googleapis/google-auth-library-php/issues/231

### CODE REVIEW

**Comment by mareksuscak** [comment]
I signed it!

**Comment by mareksuscak** [comment]
I should prolly add some tests. Let me know what you think, though.

File: src/Credentials/GCECredentials.php [hunk 1]
```diff
@@ -189,7 +189,11 @@ public static function onGce(callable $httpHandler = null)
                 // the metadata resolution was particularly slow. The latter case is
                 // "unlikely".
                 $resp = $httpHandler(
-                    new Request('GET', $checkUri),
+                    new Request(
+                        'GET',
+                        $checkUri,
+                        [GCECredentials::FLAVOR_HEADER => 'Google']
```
Comment (jdpedrie) [inline src/Credentials/GCECredentials.php]:
```suggestion
                        [self::FLAVOR_HEADER => 'Google']
```

**Comment by jdpedrie** [comment]
@googlebot I consent.

**Comment by mareksuscak** [comment]
@jdpedrie no problem and thanks for taking the time to review and contribute. I've added the unit test and made the code style consistent. Please take a look when you get a chance.

**Comment by mareksuscak** [comment]
Updated from the upstream and squashed the commits. Tests should hopefully pass now.

**Comment by jdpedrie** [comment]
Thanks again @mareksuscak!

**Pull request merged**